xonsh/completer.py

```diff
@@ -78,16 +78,54 @@ def _normpath(p):
 
     return p
 
-class Completer(object):
-    """This provides a list of optional completions for the xonsh shell."""
+class KnownCommands(object):
```

Comment:
Also agree that should live somewhere else.  tools.py would be fine, but it could also live in its own module.